File: src/main/java/nl/waterjeloen/quoridor/Direction.java
package nl.waterjeloen.quoridor;

public enum Direction {
    DOWN("down", 1, 0, false, true),
    UP("up", -1, 0, false, true),
    LEFT("left", 0, -1, true, false),
    RIGHT("right", 0, 1, true, false);

    final String name;
    final int rowDelta;
    final int columnDelta;
    final boolean isHorizontal;
    final boolean isVertical;

    Direction(String name, int rowChange, int columnChange, boolean isHorizontal, boolean isVertical) {
        this.name = name;
        this.rowDelta = rowChange;
        this.columnDelta = columnChange;
        this.isHorizontal = isHorizontal;
        this.isVertical = isVertical;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: src/main/scala/nl/zwaaltjes/quoridor/Direction.scala
package nl.zwaaltjes.quoridor

enum Direction(name: String, val rowDelta: Int, val columnDelta: Int, val isHorizontal: Boolean, val isVertical: Boolean) {
  case down extends Direction("down", rowDelta = 1, columnDelta = 0, isHorizontal = false, isVertical = true)
  case up extends Direction("up", rowDelta = -1, columnDelta = 0, isHorizontal = false, isVertical = true)
  case left extends Direction("left", rowDelta = 0, columnDelta = -1, isHorizontal = true, isVertical = false)
  case right extends Direction("right", rowDelta = 0, columnDelta = 1, isHorizontal = true, isVertical = false)

  override def toString: String =
    name
}